### DISCUSSION TITLE
Using Custom Elements

### DISCUSSION THREAD

**Discussion by LeaVerou** [Devographics/surveys#154] [State of HTML Features]
Using custom elements, that you or someone else defined.

### Code sample

```html
<sl-switch>Switch</sl-switch>
```

### MDN URL:

https://developer.mozilla.org/en-US/docs/Web/API/Web_components/Using_custom_elements

### How stable is this feature?

Stable (Widely supported for several years)

### Type of feature

HTML element

### What focus area does this feature fall under?

Consuming Web Components

### Details

Currently asked in [State of JS](https://survey.devographics.com/en-US/survey/state-of-js/2022/outline/2#custom_elements), this should probably move to State of HTML.

**Comment by joyously** [comment]
It seems to me that custom elements require JS, but not all environments that render HTML have JS.
That means that the interoperability of the content is a concern. This can easily be seen when migrating a web site that uses custom elements as part of the platform. 
Also, feed readers and mail clients don't have the JS.

**Reply by LeaVerou** [reply]
I’m not sure I follow what you are proposing for this question.

**Reply by joyously** [reply]
I'm not sure either, but I was thinking that "State of HTML" doesn't really say where and how it is used. I think the fact that content is extracted and used in multiple ways is overlooked, because everyone focuses on the browser.
So maybe it's an overall thing to keep in mind, and it shows up most often on things like custom elements.
